godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -282,6 +282,32 @@ macro_rules! impl_float_vector_glam_fns {
     };
 }
 
+/// Implements common constants and methods for integer type vectors. Works for any vector type that
+/// has `to_glam`, `from_glam` and `to_glam_real` functions.
+macro_rules! impl_integer_vector_glam_fns {
+    (
+        // Name of the vector type.
+        $Vector:ty,
+        // Type of target component, for example `real`.
+        $Scalar:ty
+    ) => {
+        impl $Vector {
+            /// Returns the length (magnitude) of this vector.
+            #[inline]
+            pub fn length(self) -> $Scalar {
+                self.to_glam_real().length()
+            }
+
+            /// Returns a new vector with each component set to 1 if it's positive, -1 if it's
+            /// negative, and 0 if it's zero.
+            #[inline]
+            pub fn sign(self) -> Self {
+                Self::from_glam(self.to_glam().signum())
+            }
```

Comment:
The implementation is wrong; `signum()` never returns 0.

Reply:
`signum` does return `0` for integers. it's only for floats `signum` is never 0. see https://doc.rust-lang.org/std/primitive.i64.html#method.signum

Reply:
Are we sure? Both IVec2::signum (i32::signum) and godot docs says it does return 0.

Reply:
I see, thanks! Now I'm even more confused why Rust chose two different behaviors here. Btw, a unit test wouldn't have raised this question in the first place 😛

Reply:
I never wrote an unit test before, only integration tests; so I don't know the workflow. I've checked multiple files but couldn't find any tests that I can figure out from. Would you mind suggesting a file with good written tests so I can do something similar? 😅

Reply:
> I see, thanks! Now I'm even more confused why Rust chose two different behaviors here. Btw, a unit test wouldn't have raised this question in the first place 😛

well signum represents the sign of a value with numbers with: 1 being positive, -1 being negative, 0 being unsigned, and NAN for NAN.

so since 0 is unsigned for integers, signum returns 0. but since floats have signed 0s, i.e +0 and -0 are distinct values, signum should return 1 or -1 respectively.